### PR TITLE
Add a note about how to handle modal dialogs.

### PR DESCRIPTION
**Pull request by HazenBabcock**
This I think adds the modal dialog handling documentation you were looking for in issue #18 and issue #19.

### CODE REVIEW

File: docs/note_dialogs.rst [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+==========================
+
+Simple Dialogs
+--------------
+
+For QMessageBox.question one approach is to ``mock`` it:
+
+.. code-block:: python
+                
+    def test_Qt(qtbot, mock):
```
Comment (The-Compiler) [inline docs/note_dialogs.rst]: This should be called `mocker`, `mock` is just a compatibility alias: https://github.com/pytest-dev/pytest-mock/issues/4

However, why require another plugin just for a doc example anyways? I think it'd be better to just use the built-in [monkeypatch](https://docs.pytest.org/en/latest/monkeypatch.html).

File: docs/note_dialogs.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+   name, age = AskNameAndAgeDialog.ask("Enter name and age because of bananas:", parent)
+   if name is not None:
+       # use name and age for bananas
+
+And now it is also easy to mock AskNameAndAgeDialog.ask when testing the form:
+
+.. code-block:: python
+                
+    def test_form_registration(qtbot, mock):
```
Comment (The-Compiler) [inline docs/note_dialogs.rst]: Same as above regarding `mock`.

**Comment by nicoddemus** [comment]
Thanks a lot @HazenBabcock!

I agree with @The-Compiler's comments so I went ahead and applied the changes myself.

Thanks again for submitting the PR, we appreciate it. 👍

**Comment by nicoddemus** [comment]
Fixed #19

**Comment by HazenBabcock** [comment]
Great, thanks for the help with this PR.

**Comment by nicoddemus** [comment]
@The-Compiler are you OK with merging this? 😁 (failures are unrelated and I plan to tackle them in #202)

**Comment by The-Compiler** [comment]
Pushed a commit to remove the last sentence as it's not true anymore. LGTM now! :+1:

**Comment by nicoddemus** [comment]
Thanks!

**Pull request merged**